online/src/app/store/reducers/cart.reducers.ts

```diff
@@ -0,0 +1,69 @@
+import { createReducer, on } from "@ngrx/store";
+import {
+    checkoutRequest, checkoutRequestError, checkoutRequestSucces, placeOrder, placeOrderSuccess, placeOrderError
+} from "../actions/cart.actions";
+import { initialCartState } from "../state/cart.state";
+
+export const cartReducer = createReducer(
+    initialCartState,
+
+    on(placeOrder, (state, { id }) => {
+        const placedOrder = state.productOrders.find(placedOrder => placedOrder.productId === id);
+        if (placedOrder === undefined) {
+            return {
+                ...state,
+                productOrders: [...state.productOrders, { productId: id, quantity: 1 }],
+                status: 'success',
+                error: "",
+            };
+        } else {
+            const addedMultipleTimesProduct = { productId: id, quantity: placedOrder.quantity + 1 }
+            return {
+                ...state,
+                productOrders: state.productOrders.map((item) => {
```

Comment:
Hint: If you want more condensed code you can skip the brackets if you only put 1 instruction after the arrow E.g. ` item => item.productId === id ? addedMultipleTimesProduct : item`.